dev_settings.py

```diff
@@ -134,6 +134,19 @@
     'icds': '100.71.184.7',
 }
 
+# corehq/apps/es/index/settings.py sets the number of replicas for the
+# case_search index to 1. But dev environments only have one node, which
+# is already in use by the primary, and so the replica cannot be
+# assigned. The following overrides those settings. (We can't use the
+# constants defined in that file, because it will cause a circular
+# import.)
+ES_SETTINGS = {
+    'case_search': {
+        'number_of_replicas': 0,
+        'number_of_shards': 5,
```

Comment:
This was something that was in my mind but never took a shot at it. Thanks @kaapstorm  for attempting it.

It seems that having multiple shards on local data may not be very practical. I am curious if the data in a local index would ever exceed 20 GBs, which is the recommended shard size for an index in Elasticsearch.

And do you think it will make sense to set these settings for all the indices?

Reply:
> I am curious if the data in a local index would ever exceed 20 GBs

Good question! I just kept adding to the data in my dev environment for years, and it grew pretty big. The case search index might have exceeded 20GB. (I created a clean environment a few months ago, so I'd need to check a backup to give you an exact figure.) I would guess there are some devs who might have more than one shard's worth of data.

Reply:
I'm on the opposite end of the spectrum from Norman in that I've never had a local environment that grew very large. I would propose that we set the number of shards to a reasonable minimum default (1?). People who are running huge local environments that need multiple shards can override the setting in `localsettings.py` if they need to.

Open questions: How would anyone who is not intimately familiar with ES in CommCare know when they need to tweak this setting? Would they become increasingly frustrated with their local experience, but be unable to troubleshoot the issue? Would it make sense to provide a troubleshooting tool (management command, for example) to propose configuration tweaks based on environment conditions?

Reply:
Great questions, @millerdev 

I don't know how one would be alerted to an issue, but I've added some instructions on how to find out how much space each index is using, how much is too much, and what to do if that happens.

Reply:
fwiw Elasticsearch says - 
> Aim for shard sizes between 10GB and 50GB

https://www.elastic.co/guide/en/elasticsearch/reference/7.17/size-your-shards.html#shard-size-recommendation